subsys/bluetooth/host/hci_ecc.c

```diff
@@ -261,7 +256,7 @@ static uint8_t le_gen_dhkey(uint8_t *key, uint8_t key_type)
 	atomic_set_bit_to(flags, USE_DEBUG_KEY,
 			  key_type == BT_HCI_LE_KEY_TYPE_DEBUG);
 
-	k_sem_give(&cmd_sem);
+	k_work_submit_to_queue(bt_long_wq_get(), &ecc_work);
```

Comment:
GATT caching is conditionally using the long WQ, but here we are unconditionally using it. Is that intentional?

Reply:
yes, since it's replacing the hci_ecc thread, we still need it if this file is compiled in.

Reply:
But will the ECC still work without the long workqueue?

Reply:
you're right that if a user puts `CONFIG_BT_LONG_WQ=n` then ECC won't work. I will add a `select` statement to `BT_TINYCRYPT_ECC` to prevent that.

Reply:
Any reason why we don't want to treat ECC and GATT caching the same? i.e. if GATT caching can be done in the system workqueue, can't ECC be that too?

Reply:
It's a good point, maybe @cvinayak could comment on that ?
I guess it could open the door to more bugs like the one this PR is trying to fix. Maybe even deadlocks ? not sure.